### ISSUE TITLE
opencv-python==4.0.1.23

### ISSUE THREAD

**Issue by d5423197** [cvg/SOLD2#51]
Hello there,

When I am trying to install all the requirements. I could not find opencv-python==4.0.1.23. Then I go to the official web, it says this version has been yanked. Is there a substitution that can be used? By the way, what is the python version recommended form this project?

Thanks,

Zhonghan Deng

**Comment by d5423197** [comment]
Hello there,

Question 1:

In step 2, you gave an example command. After step 1, we should only have a tar file which will be taken as resume_path right? Why there is an argument for checkpoint name? How to get that? 

Question 2:
In step 2, you said, "Prior to this step, make sure that the dataset config file config/wireframe_dataset.yaml has the lines gt_source_train and gt_source_test commented and you should also disable the photometric and homographic augmentations."
Does that mean we have these comments out in step 1? (prior to this step). So we need to have these set as True

Question 3:
I can just use the sold2_synthetic.tar model if I wanna skip step 1 right?

Thanks for your patience and help

Zhonghan Deng

**Comment by rpautrat** [comment]
Hi,

For the OpenCV version, we were using version 4.0.1.23 because it contained the implementations of LSD and LBD. But you should not need them to train/test SOLD2, so you can use any recent version of OpenCV, for example > 4.0.

For your other questions:
1. 'resume_path' should indeed point at the tar file that was generated in step 1. 'exp_name' refers to the name of the folder of the experiment created by step 1. I agree that it is a bit redundant, but knowing the folder name allows accessing the other training information, such as the configuration file, etc.
2. Step 1 is not using the file `config/wireframe_dataset.yaml`, so there is nothing to change there before step 1. The config file for step 1 is `config/synthetic_dataset.yaml`, and it has already the right values (i.e. with photometric and homographic augmentations set to True).
3. Yes, you should be able to use the pre-trained model to directly jump to step 2.

**Comment by d5423197** [comment]
Hello there,

Thanks for your reply.

torch 1.3.1 does not exist anymore. Which version of torch will be good to use? The code will not work if I am using torch 1.11.0.  (could you update requirements.txt if possible)

If I understand your reply correctly, at step 2, we will have photometric and homographic augmentations disabled.


Thanks,

Zhonghan Deng

**Comment by d5423197** [comment]
Hello there,

pytorch 1.3.1 require python version 3.7. under py3.7, pickle raise valueerror: unsupported pickle protocol: 5. I am confused about the version of python supposed to be used in this project.

**Comment by d5423197** [comment]
Hello there,

Could you please get a newer version of the environment you are using and update it to requirements.txt? I have tried python 3.7, 3.8, and 3.9. I  am stuck in the environment setup. The newest version of the torch is incomparable with the code. Under python 3.7,  kornia==0.2.0 is not available.

**Comment by d5423197** [comment]
Hello there,

If I get the wireframe_train_gt.h5 file from step 3. how can I visualize the line map? Is there any ready-to-use script for that? I saw the image are in size of (512, 521). But the shape of line_map is (635, 635)

**Comment by d5423197** [comment]
I have solved Question 5. I can use visualize_exported_dataset to visualize the result from step 3.

**Comment by rpautrat** [comment]
Hi,

Sorry for the late reply. I think the code can work with most Python 3 and Pytorch versions. The previous requirements were old ones, but the code is also compatible with more recent ones. I have pushed relaxed requirements now.

I have for example tested the code with **Python 3.7**, **torch 1.11.0**, and **kornia 0.6.4** and it worked all fine for me. I used pip and not anaconda, so I did not need cudatoolkit and cannot help you with that.

The error that you are mentioning seems to be coming from your torch installation, not from this code. Maybe try to completely uninstall torch and to reinstall it again (from conda or pip).

**Comment by d5423197** [comment]
Hello there,

Another quick question is:
if the base detector performs badly on some images, then there is no way that the final detector will perform much better than it.

For base detector, I mean the detector which is trained on synthetic dataset.
For final detector, I mean the detector which is trained on step 4.

**Comment by rpautrat** [comment]
Hi, yes, if the base detector is too bad, the final detector will not be able to train correctly. But the homography adaptation strategy works pretty well usually, even with a not so good base detector.

**Comment by d5423197** [comment]
Hello there,

Thanks for your reply. Is there any supervised line detection algorithm you recommend? (easy to make label). Because my custom dataset are simple bio images, there are some dark lines I want to detect.

**Comment by rpautrat** [comment]
Hi, you could try the [Line Segment Detector (LSD)](https://www.semanticscholar.org/paper/LSD%3A-a-Line-Segment-Detector-Gioi-Jakubowicz/3dceb913ccb8394e1a5abec051708f86deed5483), which works in most situations. But these detections may not be very adapted to supervise SOLD2, I have never tried it. But maybe directly using LSD could work for your application?

**Comment by d5423197** [comment]
Hello there,

Really appreciate your help. Your model did work on most of my good images. Thanks for your contribution and brilliant work. I am closing this issue now.

Best Wishes,

ZD
